File: packages/gbnf-experiment/data/20260909T151143Z_170b737d/ported_implementation/src/rules_builder/parse_space.ts
export const parse_space = (src: string, pos: number, newline_ok: boolean): number => {
  while (
    pos < src.length &&
    (src[pos] === " " ||
      src[pos] === "\t" ||
      src[pos] === "#" ||
      (newline_ok && (src[pos] === "\r" || src[pos] === "\n")))
  ) {
    if (src[pos] === "#") {
      while (pos < src.length && src[pos] !== "\r" && src[pos] !== "\n") {
        pos += 1;
      }
    } else {
      pos += 1;
    }
  }
  return pos;
};

File: packages/gbnf-experiment/data/20260909T032340Z_f01d5496/ported_implementation/gbnf/rules_builder/parse_space.py
def parse_space(src: str, pos: int, newline_ok: bool) -> int:
    def at(i: int):
        return src[i] if 0 <= i < len(src) else None

    while at(pos) in (" ", "\t", "#") or (
        newline_ok and at(pos) in ("\r", "\n")
    ):
        if at(pos) == "#":
            while at(pos) is not None and at(pos) != "\r" and at(pos) != "\n":
                pos += 1
        else:
            pos += 1
    return pos
